sklearn/ensemble/_base.py

```diff
@@ -129,23 +152,54 @@ def __init__(self, base_estimator, *, n_estimators=10, estimator_params=tuple())
     def _validate_estimator(self, default=None):
         """Check the base estimator.
 
-        Sets the base_estimator_` attributes.
+        Sets the `estimator_` attributes.
         """
-        if self.base_estimator is not None:
-            self.base_estimator_ = self.base_estimator
+        if self.estimator is not None and (
+            self.base_estimator not in [None, "deprecated"]
+        ):
```

Comment:
This is not covered by any test. Please add a test checking that this error is raised when expected.